src/notifications/api/viewsets.py

```diff
@@ -1,40 +1,78 @@
 import logging
 
-from rest_framework import viewsets
+from rest_framework import mixins, status, viewsets
+from rest_framework.parsers import JSONParser
+from rest_framework.response import Response
 from vng_api_common.permissions import ActionScopesRequired
+from vng_api_common.viewsets import CheckQueryParamsMixin
 
-from .scopes import EXAMPLE_SCOPE
-from .serializers import ExampleSerializer
+from notifications.datamodel.models import Abonnement, Kanaal
+
+from .scopes import SCOPE_NOTIF_CHANGE_ALL, SCOPE_NOTIF_READ_ALL
+from .serializers import (
+    AbonnementSerializer, KanaalSerializer, MessageSerializer
+)
 
 logger = logging.getLogger(__name__)
 
 
-class ExampleViewSet(viewsets.ModelViewSet):
-    """
-    Describe viewset.
+class AbonnementViewSet(CheckQueryParamsMixin,
+                        viewsets.ModelViewSet):
 
-    create:
-    Describe create operation.
+    queryset = Abonnement.objects.all()
+    serializer_class = AbonnementSerializer
+    lookup_field = 'uuid'
+    permission_classes = (ActionScopesRequired,)
+    required_scopes = {
+        'list': SCOPE_NOTIF_READ_ALL,
+        'retrieve': SCOPE_NOTIF_READ_ALL,
+        'create': SCOPE_NOTIF_CHANGE_ALL,
+        'destroy': SCOPE_NOTIF_CHANGE_ALL,
+        'update': SCOPE_NOTIF_CHANGE_ALL,
+        'partial_update': SCOPE_NOTIF_CHANGE_ALL,
+    }
 
-    list:
-    Describe list operation.
 
-    partial_update:
-    Describe partial_update operation.
+class KanaalViewSet(CheckQueryParamsMixin,
+                    mixins.CreateModelMixin,
+                    mixins.ListModelMixin,
+                    mixins.RetrieveModelMixin,
+                    viewsets.GenericViewSet):
 
-    destroy:
-    Describe destroy operation.
-    """
-    queryset = ...
-    serializer_class = ExampleSerializer
+    queryset = Kanaal.objects.all()
+    serializer_class = KanaalSerializer
     lookup_field = 'uuid'
+    permission_classes = (ActionScopesRequired,)
+    required_scopes = {
+        'list': SCOPE_NOTIF_READ_ALL,
+        'retrieve': SCOPE_NOTIF_READ_ALL,
+        'create': SCOPE_NOTIF_CHANGE_ALL,
+        'destroy': SCOPE_NOTIF_CHANGE_ALL,
+        'update': SCOPE_NOTIF_CHANGE_ALL,
+        'partial_update': SCOPE_NOTIF_CHANGE_ALL,
+    }
+
 
+class NotificatieViewSet(viewsets.ViewSet):
+    parser_classes = (JSONParser,)
     permission_classes = (ActionScopesRequired,)
     required_scopes = {
-        'list': EXAMPLE_SCOPE,
-        'retrieve': EXAMPLE_SCOPE,
-        'create': EXAMPLE_SCOPE,
-        'update': EXAMPLE_SCOPE,
-        'partial_update': EXAMPLE_SCOPE,
-        'destroy': EXAMPLE_SCOPE,
+        'list': SCOPE_NOTIF_READ_ALL,
+        'retrieve': SCOPE_NOTIF_READ_ALL,
+        'create': SCOPE_NOTIF_CHANGE_ALL,
+        'destroy': SCOPE_NOTIF_CHANGE_ALL,
+        'update': SCOPE_NOTIF_CHANGE_ALL,
+        'partial_update': SCOPE_NOTIF_CHANGE_ALL,
     }
+
+    def create(self, request, *args, **kwargs):
```

Comment:
Why not just define the `MessageSerializer` as serializer on this `NotificatieViewSet`?

Reply:
Ha, I thought more came out of the box with this. Oh well :)